src/rulesets/oas/functions/__tests__/oasPathParam.test.ts

```diff
@@ -233,4 +299,171 @@ describe('oasPathParam', () => {
+                in: 'path',
+                required: true,
+              },
+            ],
+          },
+        },
+      },
+    });
+
+    expect(results).toEqual([
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Parameter `boo` is not used in the path `/foo`.',
+        path: ['paths', '/foo', 'parameters', '0'],
+        severity: DiagnosticSeverity.Error,
+      }),
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Parameter `bar` is not used in the path `/foo`.',
+        path: ['paths', '/foo', 'get', 'parameters', '0'],
+        severity: DiagnosticSeverity.Error,
+      }),
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Parameter `baz` is not used in the path `/foo`.',
+        path: ['paths', '/foo', 'put', 'parameters', '0'],
+        severity: DiagnosticSeverity.Error,
+      }),
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Parameter `qux` is not used in the path `/foo`.',
+        path: ['paths', '/foo', 'put', 'parameters', '1'],
+        severity: DiagnosticSeverity.Error,
+      }),
+    ]);
+  });
+
+  test('Error if path parameter are defined multiple times', async () => {
+    const results = await s.run({
+      paths: {
+        '/foo/{boo}/{bar}/{qux}': {
+          parameters: [
+            {
+              name: 'boo',
+              in: 'path',
+              required: true,
+            },
+            {
+              name: 'boo',
+              in: 'path',
+              required: true,
+            },
+            {
+              name: 'bar',
+              in: 'path',
+              required: true,
+            },
+          ],
+          get: {
+            parameters: [
+              {
+                name: 'bar',
+                in: 'path',
+                required: true,
+              },
+              {
+                name: 'qux',
+                in: 'path',
+                required: true,
+              },
+            ],
+          },
+          put: {
+            parameters: [
+              {
+                name: 'qux',
+                in: 'path',
+                required: true,
+              },
+              {
+                name: 'qux',
+                in: 'path',
+                required: true,
+              },
+            ],
+          },
+        },
+      },
+    });
+
+    expect(results).toEqual([
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Path parameter `boo` is defined multiple times. Path parameters must be unique.',
+        path: ['paths', '/foo/{boo}/{bar}/{qux}', 'parameters', '1'],
+        severity: DiagnosticSeverity.Error,
+      }),
+      expect.objectContaining({
+        code: 'path-params',
+        message: 'Path parameter `bar` is defined multiple times. Path parameters must be unique.',
+        path: ['paths', '/foo/{boo}/{bar}/{qux}', 'get', 'parameters', '0'],
```

Comment:
@P0lip This duplicated parameter (at path and operation level) wasn't previously detected